rust/examples/read_csv.rs

```diff
@@ -0,0 +1,79 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+extern crate arrow;
+
+use arrow::array::{ListArray, PrimitiveArray};
+use arrow::csvreader::CsvFile;
+use arrow::datatypes::{DataType, Field, Schema};
+use std::fs::File;
+use std::sync::Arc;
+
+fn main() {
+    let schema = Schema::new(vec![
+        Field::new("city", DataType::Utf8, false),
+        Field::new("lat", DataType::Float64, false),
+        Field::new("lng", DataType::Float64, false),
+    ]);
+
+    let file = File::open("test/data/uk_cities.csv").unwrap();
+
+    let mut csv = CsvFile::open(file, Arc::new(schema), false, 1024, None);
+    let batch = csv.next().unwrap().unwrap();
+
+    println!(
+        "Loaded {} rows containing {} columns",
+        batch.num_rows(),
+        batch.num_columns()
+    );
+
+    let city = batch
+        .column(0)
+        .as_any()
+        .downcast_ref::<ListArray>()
+        .unwrap();
+    let lat = batch
+        .column(1)
+        .as_any()
+        .downcast_ref::<PrimitiveArray<f64>>()
+        .unwrap();
+    let lng = batch
+        .column(2)
+        .as_any()
+        .downcast_ref::<PrimitiveArray<f64>>()
```

Comment:
Are we sure that the alignment requires are met if we simple downcast from any?  Alignments are checked through our existing constructors but would not be checked if we downcast, right?

Reply:
I have no idea to be honest. I am definitely looking for guidance here on how to access these arrays after they are built.

Reply:
Actually, I think it's fine as you are using our builders internally.  I'll take a closer look when I get a chance.

Thanks @andygrove